lib/VM/JSLib/TypedArray.cpp

```diff
@@ -1746,6 +1746,53 @@ typedArrayPrototypeToLocaleString(void *, Runtime &runtime, NativeArgs args) {
   return HermesValue::encodeStringValue(*builder->getStringPrimitive());
 }
 
+/// ES14.0 23.2.3.32
+CallResult<HermesValue>
+typedArrayPrototypeToReversed(void *, Runtime &runtime, NativeArgs args) {
+  GCScope gcScope{runtime};
+
+  // 2. Perform ? ValidateTypedArray(O).
+  if (JSTypedArrayBase::validateTypedArray(runtime, args.getThisHandle()) ==
+      ExecutionStatus::EXCEPTION) {
+    return ExecutionStatus::EXCEPTION;
+  }
+
+  // 1. Let O be this value
+  auto self = args.vmcastThis<JSTypedArrayBase>();
+
+  // 3. Let len be O.[[ArrayLength]].
+  JSArrayBuffer::size_type len = self->getLength();
+
+  auto byteWidth = self->getByteWidth();
+
+  // 4. Let A be ? TypedArrayCreateSameType(O, « 𝔽(len) »).
+  auto aRes = JSTypedArrayBase::allocateSpecies(runtime, self, len);
+  if (LLVM_UNLIKELY(aRes == ExecutionStatus::EXCEPTION)) {
+    return ExecutionStatus::EXCEPTION;
+  }
+  auto A = aRes.getValue();
+
+  // 5. Let k be 0.
+  JSArrayBuffer::size_type k = 0;
+
+  auto aBuffer = A->getBuffer(runtime);
+  auto srcBuffer = self->getBuffer(runtime);
+
+  // 6. Repeat, while k < len,
+  while (k < len) {
+    // 6a. Let from be ! ToString(𝔽(length - k - 1)).
+    JSArrayBuffer::size_type from = len - k - 1;
+
+    // 6d. Perform ! Set(A, Pk, fromValue, true).
+    JSArrayBuffer::copyDataBlockBytes(runtime, aBuffer, k * byteWidth, srcBuffer, from * byteWidth, byteWidth);
```

Comment:
Sure thing! Just wanted to first check if the current approach is corrrect :)

Reply:
WDYT about creating a static utility function for copying elements in `JSTypedArray` (where byte width is known) rather than copying bytes?

Reply:
Then you have to specialize it for every valid width. Probably not necessary at the moment. It can be done later if we do need to dig more performance. For now, `memcpy` on the underlying `data_` array is good enough.